Specs/Core/GregorianDateSpec.js

```diff
@@ -0,0 +1,300 @@
+      expect(minimumDate.second).toEqual(0);
+      expect(minimumDate.millisecond).toEqual(0);
+      expect(minimumDate.isLeapSecond).toBeFalsy();
+    });
+
+    it("Constructs the minimum possible time of the day when only year, month, day, hour and minutes parameters are passed", function () {
+      const minimumDate = new GregorianDate(2022, 2, 28, 10, 59);
+      expect(minimumDate.year).toEqual(2022);
+      expect(minimumDate.month).toEqual(2);
+      expect(minimumDate.day).toEqual(28);
+      expect(minimumDate.hour).toEqual(10);
+      expect(minimumDate.minute).toEqual(59);
+      expect(minimumDate.second).toEqual(0);
+      expect(minimumDate.millisecond).toEqual(0);
+      expect(minimumDate.isLeapSecond).toBeFalsy();
+    });
+
+    it("Constructs the minimum possible time of the day when only year, month, day, hour, minutes and seconds parameters are passed", function () {
+      const minimumDate = new GregorianDate(2022, 2, 28, 10, 59, 59);
+      expect(minimumDate.year).toEqual(2022);
+      expect(minimumDate.month).toEqual(2);
+      expect(minimumDate.day).toEqual(28);
+      expect(minimumDate.hour).toEqual(10);
+      expect(minimumDate.minute).toEqual(59);
+      expect(minimumDate.second).toEqual(59);
+      expect(minimumDate.millisecond).toEqual(0);
+      expect(minimumDate.isLeapSecond).toBeFalsy();
+    });
+
+    it("Constructs the date with leap second", function () {
+      const minimumDate = new GregorianDate(2022, 2, 28, 10, 59, 60, 100, true);
+      expect(minimumDate.year).toEqual(2022);
+      expect(minimumDate.month).toEqual(2);
+      expect(minimumDate.day).toEqual(28);
+      expect(minimumDate.hour).toEqual(10);
+      expect(minimumDate.minute).toEqual(59);
+      expect(minimumDate.second).toEqual(60);
+      expect(minimumDate.millisecond).toEqual(100);
+      expect(minimumDate.isLeapSecond).toBeTruthy();
+    });
+  });
+  describe("Negative scenarios", function () {
+    it("Should throw error if invalid year is passed", function () {
+      expect(function () {
+        return new GregorianDate(-1, 2, 4, 23, 54, 0, 999.9, false);
+      }).toThrowDeveloperError();
+      expect(function () {
+        return new GregorianDate(0, 2, 4, 23, 54, 0, 999.9, false);
+      }).toThrowDeveloperError();
+      expect(function () {
+        return new GregorianDate(10000, 2, 4, 23, 54, 0, 999.9, false);
+      }).toThrowDeveloperError();
+    });
+
+    it("Should throw error if invalid month is passed", function () {
+      expect(function () {
+        return new GregorianDate(2022, -1, 4, 23, 54, 0, 999.9, false);
+      }).toThrowDeveloperError();
+      expect(function () {
+        return new GregorianDate(2022, 0, 4, 23, 54, 0, 999.9, false);
+      }).toThrowDeveloperError();
+      expect(function () {
+        return new GregorianDate(2022, 13, 4, 23, 54, 0, 999.9, false);
+      }).toThrowDeveloperError();
+    });
+
+    it("Should throw error if invalid day is passed", function () {
```

Comment:
For easier readability can we split this up into three tests?
- "Should throw error if invalid day is passed," which handles invalid values (-10, 0, 32)
- "Should throw error if day is out of range for month," which handles cases where months with 30 days are given an input of 31, and where February is given an input of 30
- "Should throw error if leap day is given for non-leap year", which handles the special February 29 case

Organizing them this way will help us remember the intentions of each test for future reference